dao/src/main/java/org/wso2/testgrid/dao/repository/DeploymentPatternRepository.java

```diff
@@ -27,82 +27,85 @@
  */
-public class DatabaseRepository extends AbstractRepository<Database> {
+public class DeploymentPatternRepository extends AbstractRepository<DeploymentPattern> {
 
     /**
      * Constructs an instance of the repository class.
      *
      * @param entityManager {@link EntityManager} instance
      */
-    public DatabaseRepository(EntityManager entityManager) {
+    public DeploymentPatternRepository(EntityManager entityManager) {
         super(entityManager);
     }
 
     /**
-     * Persists an {@link Database} instance in the database.
+     * Persists an {@link DeploymentPattern} instance in the database.
      *
-     * @param entity database to persist in the database
-     * @return added or updated {@link Database} instance
-     * @throws TestGridDAOException thrown when error on persisting the database instance
+     * @param entity TestPlan to persist in the database
+     * @return added or updated {@link DeploymentPattern} instance
+     * @throws TestGridDAOException thrown when error on persisting the TestPlan instance
      */
-    public Database persist(Database entity) throws TestGridDAOException {
+    public DeploymentPattern persist(DeploymentPattern entity) throws TestGridDAOException {
         return super.persist(entity);
     }
 
     /**
-     * Removes an {@link Database} instance from database.
+     * Removes an {@link DeploymentPattern} instance from database.
      *
-     * @param entity database instance to be removed from database.
+     * @param entity TestPlan instance to be removed from database.
      * @throws TestGridDAOException thrown when error on removing entry from database
      */
-    public void delete(Database entity) throws TestGridDAOException {
+    public void delete(DeploymentPattern entity) throws TestGridDAOException {
         super.delete(entity);
+        entity.setTestPlans(null);
     }
 
     /**
-     * Find a specific {@link Database} instance from database for the given primary key.
+     * Find a specific {@link DeploymentPattern} instance from database for the given primary key.
      *
      * @param id primary key of the entity to be searched for
-     * @return instance of an {@link Database} matching the given primary key
+     * @return instance of an {@link DeploymentPattern} matching the given primary key
      * @throws TestGridDAOException thrown when error on searching for entity
      */
-    public Database findByPrimaryKey(String id) throws TestGridDAOException {
-        return findByPrimaryKey(Database.class, id);
+    public DeploymentPattern findByPrimaryKey(String id) throws TestGridDAOException {
+        return findByPrimaryKey(DeploymentPattern.class, id);
     }
 
     /**
-     * Returns a list of {@link Database} instances matching the given criteria.
+     * Returns a list of {@link DeploymentPattern} instances matching the given criteria.
      *
      * @param params parameters (map of field name and values) for obtaining the result list
```

Comment:
Do we need "params parameters" in the comment? I assume "parameters" would be sufficient.